### DISCUSSION TITLE
Compressing JPGs which method should I use ?

### DISCUSSION THREAD

**Discussion by trevski** [dlemstra/Magick.NET#1234] [Help]
So I am using 
Dim optimizer As New ImageMagick.ImageOptimizers.JpegOptimizer()

         If bLossless Then
                            optimizer.LosslessCompress(imgfileInfo)
                        Else
                            optimizer.OptimalCompression = True
                            optimizer.Progressive = True                            
                                optimizer.Compress(imgfileInfo)                        
                        End If

Neither give me any reduction in filesize however when I use

             Using mi As ImageMagick.MagickImage = New ImageMagick.MagickImage(imgfileInfo.FullName, ImageMagick.MagickFormat.Jpg)
                                mi.Format = mi.Format ' Get Or Set the format Of the image.
                            mi.Quality = 75
                            mi.Write(imgfileInfo.FullName)
                            End Using

I get a significant drop in file-size

**Comment by dlemstra** [comment]
It all depends on what you want to do with your images. If you change the quality you will change how the image looks. And the ImageOptimizers will not change how your image looks so this means that the size will probably be larger then when you change the quality and re-encode the image.

**Reply by jdang67** [reply]
Can someone write this component and share some sample code on how to perform image quality reduction?
